craft_parts/parts.py

```diff
@@ -642,6 +639,23 @@ def part_has_overlay(data: Dict[str, Any]) -> bool:
     return spec.has_overlay
 
 
+def part_has_slices(data: Dict[str, Any]) -> bool:
```

Comment:
I think it's important to add a basic test for `part_has_slices()` and `is_slice()`. They are currently covered by the existing validator test but if refactor the validator the functions might become uncovered